Shopist/Shopist/Scenes/CartViewController.swift

```diff
@@ -56,9 +81,37 @@ final class CartViewController: UITableViewController {
     }
 
     @objc private func pay() {
-        print("Paid")
-        cart.products.removeAll()
-        dismissPage()
+        if let randomError = Self.randomError {
+            self.handleError(randomError)
+            return
+        }
+        api.checkout(with: discountField.text) { result in
+            switch result {
+            case .success(let value):
+                self.handleSuccess()
+            case .failure(let error):
+                self.handleError(error)
+            }
+        }
+    }
+
+    private func handleSuccess() {
+        DispatchQueue.main.async {
+            cart.products.removeAll()
+            (self.presentingViewController as? UINavigationController)?.popViewController(animated: false)
+            self.dismissPage()
```

Comment:
When thinking about UITests automation (to play Shopist scenarios), maybe it would be easier to also present a success alert from here? Then we'll be just asserting the alert content, not hierarchy change.